### PR TITLE
feat(Execute Workflow Trigger Node): Add types for input fields

### PR DESCRIPTION
**Pull request by CharlieKolb**
## Summary

Support user-defined types for fields and expose related options.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-2909/execute-workflow-trigger-node

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts [hunk 1]
```diff
@@ -78,41 +86,95 @@ export class ExecuteWorkflowTrigger implements INodeType {
...
+									{
+										name: 'Array',
+										value: 'array',
+									},
+									{
+										name: 'Object',
+										value: 'object',
+									},
+									// Intentional omission of `dateTime`, `time`, `string-alphanumeric`, `form-fields`, `jwt` and `url`
+								] as ValueOptions[],
```
Comment (igatanasov) [inline packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts:125]: nice

**Review by igatanasov — APPROVED** [review]
Hi @CharlieKolb ,

LGTM. I have left 2 minor commnets.

File: packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts [hunk 2]
```diff
@@ -5,11 +5,18 @@ import {
...
+	type FieldType,
 } from 'n8n-workflow';
 
 const WORKFLOW_INPUTS = 'workflowInputs';
+const INPUT_OPTIONS = 'inputOptions';
 const VALUES = 'values';
 
+type ValueOptions = { name: string; value: FieldType };
+
+const DEFAULT_PLACEHOLDER = null;
```
[the remark this answers is not in the document]
Comment (CharlieKolb) [reply packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts:18]: We'll do null for now, but may circle back to this later.

File: packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts [hunk 3]
```diff
@@ -5,11 +5,18 @@ import {
 	type IExecuteFunctions,
 	type INodeType,
 	type INodeTypeDescription,
+	validateFieldType,
+	type FieldType,
 } from 'n8n-workflow';
 
 const WORKFLOW_INPUTS = 'workflowInputs';
```
[the remark this answers is not in the document]
Comment (CharlieKolb) [reply packages/nodes-base/nodes/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts:12]: Let's do that post merge on your feature branch, great idea 🙌

**Pull request merged**